src/bci_build/package/golang/README.md.j2

```diff
@@ -0,0 +1,71 @@
+# {{ image.pretty_name }} Container Image
+
+{% include 'badges.j2' %}
+
+## Description
+
+[Go](https://go.dev/) (a.k.a., Golang) is a programming language first developed at Google. It is a statically-typed language with syntax loosely derived from C, but with additional features such as garbage collection, type safety, some dynamic-typing capabilities, additional built-in types (e.g., variable-length arrays and key-value maps), and a large standard library.
+
+## Usage
+
+The most straightforward way to use this image is to use a Go container as both the build and runtime environment.
+
+To compile and deploy your application, copy the sources, fetch dependencies (assuming go.mod for dependency management), and build the binary:
+
+(assuming it uses go.mod for dependency management):
+
+```Dockerfile
+FROM {{ image.pretty_reference }}
+
+WORKDIR /app
+
+COPY . .
+
+RUN go mod download && go mod verify
+RUN go build -v -o /usr/local/bin/app ./...
+
+CMD ["app"]
+```
```

Comment:
We really shouldn't advertise the usage of the go container for runtime, it is enormous (> 500MB uncompressed!). We should be advertising either to deploy the app via scratch or at least bci-base, bci-minimal, bci-micro or bci-busybox. But an image build as described here will be giant, as this process leaves not only the go compiler, but also all the sources, dependencies & build results in the image.

Reply:
Ok, that makes sense, not sure which image is better, or if it is better to not mention any images at all, and just say this image is for development only.

Reply:
I reworked this part, take a look.